plugins/bufferlines/barbar.nix

```
@@ -69,249 +24,592 @@ let
+      Enables highlighting of visible buffers.
     '';
 
-    maximumPadding =
-      helpers.defaultNullOpts.mkInt 4
-        "Sets the maximum padding width with which to surround each tab";
+    icons = {
+      buffer_index =
+        helpers.defaultNullOpts.mkNullableWithRaw
+          (
+            with types;
+            either bool (enum [
+              "superscript"
+              "subscript"
+            ])
+          )
+          false
+          "If `true`, show the index of the buffer with respect to the ordering of the buffers in the tabline.";
+
+      buffer_number = helpers.defaultNullOpts.mkNullableWithRaw (
+        with types;
+        either bool (enum [
+          "superscript"
+          "subscript"
+        ])
+      ) false "If `true`, show the `bufnr` for the associated buffer.";
+
+      button = helpers.defaultNullOpts.mkNullableWithRaw (with types; either str (enum [ false ])) "" ''
+        The button which is clicked to close / save a buffer, or indicate that it is pinned.
+        Use `false` to disable it.
+      '';
 
-    minimumPadding =
-      helpers.defaultNullOpts.mkInt 1
-        "Sets the minimum padding width with which to surround each tab";
+      diagnostics = mkOption {
+        type = types.submodule {
+          freeformType = with types; attrsOf anything;
+          options =
+            let
+              mkDiagnosticIconOptions = iconDefault: {
+                enabled = helpers.defaultNullOpts.mkBool false ''
+                  Enable showing diagnostics of this `|diagnostic-severity|` in the 'tabline'.
+                '';
+
+                icon = helpers.defaultNullOpts.mkStr iconDefault ''
+                  The icon which accompanies the number of diagnostics.
+                '';
+              };
+            in
+            {
+              "vim.diagnostic.severity.ERROR" = mkDiagnosticIconOptions " ";
+              "vim.diagnostic.severity.HINT" = mkDiagnosticIconOptions "󰌶 ";
+              "vim.diagnostic.severity.INFO" = mkDiagnosticIconOptions " ";
+              "vim.diagnostic.severity.WARN" = mkDiagnosticIconOptions " ";
+            };
+        };
+        apply = helpers.toRawKeys;
+        default = { };
+        description =
+          helpers.defaultNullOpts.mkDesc
+            {
+              "vim.diagnostic.severity.ERROR" = {
+                enabled = false;
+                icon = " ";
+              };
+              "vim.diagnostic.severity.HINT" = {
+                enabled = false;
+                icon = "󰌶 ";
+              };
+              "vim.diagnostic.severity.INFO" = {
+                enabled = false;
+                icon = " ";
+              };
+              "vim.diagnostic.severity.WARN" = {
+                enabled = false;
+                icon = " ";
+              };
+            }
+            ''
+              Set the icon for each diagnostic level.
+              The keys will be automatically translated to raw lua:
```

I'd add a paragraph break, personally:
```suggestion

              The keys will be automatically translated to raw lua:
```